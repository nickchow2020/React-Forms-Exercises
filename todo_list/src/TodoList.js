import React,{useState} from "react"
import NewTodoForm from "./NewTodoForm"
import Todo from "./Todo"
import { v4 as uuid } from 'uuid';

import "./TodoList.css"

const TodoList = ()=>{

    const [todos,setTodo] = useState([])

    const addTodo = (todo)=>{
        setTodo(todos => ([...todos,todo]))
    }

    const removeTodo = (todo)=>{
        setTodo(todos => (
            todos.filter(tdo => tdo !== todo)
        ))
    }

    return (
        <div className="todoWrap">
            <div>
                <NewTodoForm addTodo={addTodo}/>
                <div>
                    {
                        todos.map(todo => <Todo 
                        todo={todo} 
                        key={uuid()}  
                        removeTodo={removeTodo}/>)
                    }
                </div>
            </div>
        </div>
    )
}


export default TodoList