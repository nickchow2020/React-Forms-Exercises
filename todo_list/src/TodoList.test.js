import { render,fireEvent, getByText} from '@testing-library/react';
import TodoList from './TodoList';

test("should it render",()=>{
    render(<TodoList />)
})

test("should it match it's snapshot",()=>{
    const {asFragment} = render(<TodoList />)
    expect(asFragment()).toMatchSnapshot()
})

test("should it add new todo",()=>{
    const {getByTestId,getByText} = render(<TodoList />)

    const todoInput = getByTestId("todo")
    const todoSumit = getByTestId("addTodo")

    fireEvent.change(todoInput,{target:{value:"doing homework"}})
    fireEvent.click(todoSumit)

    expect(getByText("doing homework")).toBeInTheDocument()
})



test("should it delete todo",()=>{
    const {getByTestId,getByText} = render(<TodoList />)

    const todoInput = getByTestId("todo")
    const todoSumit = getByTestId("addTodo")

    fireEvent.change(todoInput,{target:{value:"doing homework"}})
    fireEvent.click(todoSumit)

    const deleteButton = getByText("X")

    expect(deleteButton).toBeInTheDocument()

    fireEvent.click(deleteButton)

    expect(deleteButton).not.toBeInTheDocument()

})



