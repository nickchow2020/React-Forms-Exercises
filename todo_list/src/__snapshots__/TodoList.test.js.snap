// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should it match it's snapshot 1`] = `
<DocumentFragment>
  <div
    class="todoWrap"
  >
    <div>
      <form>
        <h1>
          Add New Todo
        </h1>
        <label>
          <input
            data-testid="todo"
            id="todo"
            name="todo"
            placeholder="add todo"
            value=""
          />
        </label>
        <button
          data-testid="addTodo"
        >
          Add Todo
        </button>
      </form>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
